picard/script.py

```diff
@@ -910,6 +910,143 @@ def func_is_video(parser):
+    """
+    return text[::-1]
+
+
+def func_substr(parser, text, start_index, end_index):
+    """Extract a specified portion of a string.
+
+    Arguments:
+        text {str} -- The string from which the extract will be made.
+        start_index {int} -- Index of the first character to extract.
+        end_index {int} -- Index of the first character that will not be extracted.
+
+    Returns:
+        str -- Returns the substring beginning with the character at the start index,
+               up to (but not including) the character at the end index.  The first
+               character is at index number 0.  If the start index is left blank, it
+               defaults to the first character in the string.  If the end index is
+               left blank, it defaults to the number of characters in the string.
+               If either index is negative, it is subtracted from the total number of
+               characters in the string to provide the index used.
+    """
+    _start = int(start_index) if start_index else None
+    _end = int(end_index) if end_index else None
+    return text[_start:_end]
+
+
+def func_get_multi_item(parser, multi, item_index, separator=MULTI_VALUED_JOINER):
+    """Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+
+    Arguments:
+        multi {multi-value} -- The multi-value from which the item is to be retrieved.
+        item_index {int} -- The zero-based index of the item to be retrieved.
+
+    Returns:
+        str -- Returns value of the item at the specified index in the multi-value variable.  Index values are zero-based.
+    """
+    if item_index:
+        _index = int(item_index.eval(parser))
+    else:
+        return ''
+    _multi = _get_multi_values(parser, multi, separator)
+    try:
+        return str(_multi[_index])
+    except IndexError:
+        return ''
+
+
+def func_foreach_multi_item(parser, multi, loop_code, _loop_value="_loop_value", _loop_count="_loop_count", separator=MULTI_VALUED_JOINER):
+    """Iterates over each element found in the specified multi-value variable, executing the specified code.
+    For each loop, the element value is first stored in the tag specified by _loop_value and the count is
+    stored in the tag specified by _loop_count.  This allows the element or count value to be accessed within
+    the code script.
+
+    Arguments:
+        multi {multi-value} -- The multi-value to be iterated.
+        loop_code {str} -- Script code to be processed on each iteration.
+
+    Keyword Arguments:
+        _loop_value {str} -- Tag name to use for the element for the current loop (default: {"_loop_value"})
+        _loop_count {str} -- Tag name to use for the count of the current loop (default: {"_loop_count"})
+    """
+    loop_count = 0
+    _multi = _get_multi_values(parser, multi, separator)
+    for value in _multi:
+        loop_count += 1
+        func_set(parser, _loop_count, str(loop_count))
```

Comment:
`loop_count` and `_loop_count` ?
Using '_' prefix is mainly for private functions, or unused variables (though that's just a convention).
Also you may want to use `enumerate()`.

Reply:
Underscores removed.  I'll look at `enumerate()` (didn't notice that comment until after I submitted my update)..

Reply:
Switched to use `enumerate()`.